framework/include/utils/RankTwoTensor.h

```diff
@@ -169,10 +193,23 @@ class RankTwoTensorTempl : public TensorValue<T>
    */
   void fillFromScalarVariable(const VariableValue & scalar_variable);
 
-public:
   /// returns _coords[i][c], ie, column c, with c = 0, 1, 2
   TypeVector<T> column(const unsigned int c) const;
 
+  /// multiply vector v with row n of this tensor
+  T rowMultiply(std::size_t n, const TypeVector<T> & v) const;
+
+  /// return the matrix multiplied with its transpose A*A^T (guaranteed symmetric)
+  static RankTwoTensorTempl<T> timesTranspose(const RankTwoTensorTempl<T> &);
+
+  /// return the matrix plus its transpose A+A^T (guaranteed symmetric)
+  static RankTwoTensorTempl<T> plusTranspose(const RankTwoTensorTempl<T> &);
+
+  /**
+   * Returns the matrix squared
+   */
+  RankTwoTensorTempl<T> sqr() const;
```

Comment:
Could we call this `sq` instead of `sqr`, because I think `sqr` might be confused with square root.

Reply:
how about we just call it `square` to make it even clearer?

Reply:
It's your call :)